tests/phpunit/integration/Modules/AdsTest.php

```diff
@@ -21,46 +23,97 @@
  */
 class AdsTest extends TestCase {
 
+	/**
+	 * Ads object.
+	 *
+	 * @var Ads
+	 */
+	private $ads;
+
+	/**
+	 * Plugin context.
+	 *
+	 * @var Context
+	 */
+	private $context;
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->context = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$this->ads     = new Ads( $this->context );
+	}
+
 	public function test_magic_methods() {
-		$ads = new Ads( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+		$this->assertEquals( 'ads', $this->ads->slug );
+		$this->assertEquals( 'Ads', $this->ads->name );
+		$this->assertEquals( 'https://google.com/ads', $this->ads->homepage );
+	}
 
-		$this->assertEquals( 'ads', $ads->slug );
-		$this->assertEquals( 'Ads', $ads->name );
-		$this->assertEquals( 'https://google.com/ads', $ads->homepage );
+	public function test_register() {
+		remove_all_actions( 'template_redirect' );
+		remove_all_filters( 'googlesitekit_inline_modules_data' );
+
+		$this->ads->register();
+
+		$this->assertTrue( has_action( 'template_redirect' ) );
+		$this->assertTrue( has_filter( 'googlesitekit_inline_modules_data' ) );
 	}
 
 	public function test_is_connected_when_ads_conversion_id_is_set() {
-		$ads = new Ads( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+		$this->assertFalse( $this->ads->is_connected() );
 
-		$this->assertFalse( $ads->is_connected() );
+		$this->ads->get_settings()->set( array( 'conversionID' => 'AW-123456789' ) );
 
-		$ads->get_settings()->set( array( 'conversionID' => 'AW-123456789' ) );
+		$this->assertTrue( $this->ads->is_connected() );
+	}
 
-		$this->assertTrue( $ads->is_connected() );
+	public function test_inline_data_initial_state__module_not_connected() {
+		$inline_modules_data = apply_filters( 'googlesitekit_inline_modules_data', array() );
+
+		$this->assertArrayNotHasKey( 'ads', $inline_modules_data );
 	}
 
-	public function test_settings_reset_on_deactivation() {
-		$ads = new Ads( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+	public function test_inline_custom_dimension_data_initial_state__module_connected() {
+		$this->ads->register();
 
-		$ads->get_settings()->set( array( 'conversionID' => 'AW-123456789' ) );
+		// Ensure the module is connected.
+		$options = new Options( $this->context );
+		$options->set(
+			Settings::OPTION,
+			array(
+				'conversionID' => 'AW-12345',
+			)
+		);
 
-		$ads->on_deactivation();
+		$inline_modules_data = apply_filters( 'googlesitekit_inline_modules_data', array() );
 
-		$ads_settings = $ads->get_settings()->get();
+		$this->assertEquals(
+			array(
+				'supportedConversionEvents' => array(),
+			),
+			$inline_modules_data['ads']
+		);
+	}
+
+	public function test_settings_reset_on_deactivation() {
```

Comment:
For PHPUnit, we generally have the following naming convention for testing. I realize this wasn't entirely followed previously but please try to maintain it going forward.

- `function test_{method_name}` and
- `function test_{method_name}__{variant}`